tangram_bundler/__init__.py

```diff
@@ -1,97 +1,231 @@
+        if 'texture' in stylesNode[style]:
+            stylesNode[style]['texture'] = resolveGenericPath(stylesNode[style]['texture'], basePath)
+        if 'material' in stylesNode[style]:
+            resolveMaterialTextureUrls(stylesNode[style]['materials'], basePath)
+        if 'shaders' in stylesNode[style]:
+            resolveShaderTextureUrls(stylesNode[style]['shaders'], basePath)
+
+def resolveSceneFontsUrl(fontsNode, basePath):
+    if (type(fontsNode) is dict):
+        for font in fontsNode:
+            fontNode = fontsNode[font]
+            if (type(fontNode) is dict and 'url' in fontNode):
+                fontNode['url'] = resolveGenericPath(fontNode['url'], basePath)
+            elif(type(fontNode) is list):
+                for f in fontNode:
+                    if 'url' in f:
+                        f['url'] = resolveGenericPath(f['url'], basePath)
+
+def resolveSceneUrls(yamlRoot, filename):
+    basePath = filename
+    if 'textures' in yamlRoot:
+        resolveSceneTextureUrls(yamlRoot['textures'], basePath)
+    if 'styles' in yamlRoot:
+        resolveSceneStyleUrls(yamlRoot['styles'], basePath)
+    if 'fonts' in yamlRoot:
+        resolveSceneFontsUrl(yamlRoot['fonts'], basePath)
+
+
+def importSceneRecursive(yamlRoot, filename, allImports, importedScenes):
+    if filename in importedScenes:
+        return
+    importedScenes.add(filename)
+    sceneNode = allImports[filename]
+    if ((sceneNode is None) or (type(sceneNode) is not dict)):
+        return
+    imports = getSceneImports(filename)
+    sceneNode['import'] = None
+    for importScene in imports:
+        importSceneRecursive(yamlRoot, importScene, allImports, importedScenes)
+    importedScenes.remove(filename)
+    print "merging scene file to root: ", filename
+    mergeMapFields(yamlRoot, sceneNode)
+    resolveSceneUrls(yamlRoot, filename)
 
 # ================================== Main functions
-def bundler(full_filename):
-    print full_filename, os.getcwd()
-    filename, file_extension = os.path.splitext(full_filename)
-    if file_extension == '.yaml':
-        all_dependencies = []
-
-        # 1st order dependencies
-        addDependencies(all_dependencies, './'+full_filename)
-
-        # 2nd order theme dependencies
-        try:
-            for file in os.listdir(os.getcwd() + "/themes"):
-                if file.endswith(".yaml"):
-                    # track like normal
-                    all_dependencies.append("themes/" + file)
-                    # some themes require additional assets
-                    print "looking at dependencies for: %s" % (file,)
-                    addDependencies(all_dependencies, "themes/" + file)
-        except Exception:
-            print "\tskipping: themes (none found)"
-
-        files = list(set(all_dependencies))
+def bundler(fullFilename):
+    print fullFilename, os.getcwd()
+    filename, fileExtension = os.path.splitext(fullFilename)
+    if fileExtension == '.yaml':
```

Comment:
Should probably at least accept `.yml` too. Strictly speaking we don't need to care about the file extension at all.